src/actions/collapseHoverDown.ts

```diff
@@ -0,0 +1,197 @@
+import DropThoughtZone from '../@types/DropThoughtZone'
+import Path from '../@types/Path'
+import State from '../@types/State'
+import Thunk from '../@types/Thunk'
+import Timer from '../@types/Timer'
+import { EXPAND_HOVER_DELAY, HOME_PATH } from '../constants'
+import equalPath from '../util/equalPath'
+import hashPath from '../util/hashPath'
+import isDescendantPath from '../util/isDescendantPath'
+
+let collapseDownTimer: Timer | null = null
+
+// Track the previous hovering path to detect changes
+let previousHoveringPath: Path | null = null
+
+/** Clears active delayed dispatch. */
+const clearTimer = () => {
+  if (collapseDownTimer) {
+    clearTimeout(collapseDownTimer)
+    collapseDownTimer = null
+  }
+}
+
+/** Delays dispatch of collapseHoverDown. */
+const collapseHoverDownDebounced =
+  (path: Path): Thunk =>
+  dispatch => {
+    clearTimer()
+    collapseDownTimer = setTimeout(() => {
+      dispatch({ type: 'collapseHoverDown', path })
+      collapseDownTimer = null
+    }, EXPAND_HOVER_DELAY)
+  }
+
+/**
+ * Removes a path and all its descendants from expandHoverDownPaths and expanded states.
+ * This ensures that when a parent is collapsed, all its children are collapsed as well.
+ */
+const collapseHoverDown = (state: State, { path }: { path: Path }): State => {
+  // Get all paths from both expanded and expandHoverDownPaths states
+  const expandedPaths = Object.entries(state.expanded).map(([_, path]) => path)
```

Comment:
FYI We have [keyValueBy](https://github.com/cybersemics/em/blob/37f3e98916904810830c18b92436cad2016fac47/src/util/keyValueBy.ts) to filter objects without having to map out the entries and perform mutations on temporary data structures.

Reply:
Oh yep you are right! Thanks for the information